src/ProviderImplementations/OpenAi/OpenAiImageGenerationModel.php

```diff
@@ -4,48 +4,386 @@
     }
 
     /**
-     * @inheritDoc
+     * Prepares the given prompt and the model configuration into parameters for the API request.
+     *
+     * @since n.e.x.t
+     *
+     * @param list<Message> $prompt The prompt to generate an image for. Should be a single user message.
+     * @return array<string, mixed> The parameters for the API request.
      */
     protected function prepareGenerateImageParams(array $prompt): array
     {
-        $params = parent::prepareGenerateImageParams($prompt);
+        $config = $this->getConfig();
+        $modelId = $this->metadata()->getId();
+
+        // The Responses API with image_generation tool requires a model that supports it.
+        // We use a capable model like gpt-4o to process the request with the image_generation tool.
+        $params = [
+            'model' => $this->getHostModelForImageGeneration($modelId),
+            'input' => $this->preparePromptParam($prompt),
+            'tools' => [
+                $this->prepareImageGenerationTool($modelId),
+            ],
+        ];
 
         /*
-         * Only the newer 'gpt-image-' models support passing a MIME type ('output_format').
-         * Conversely, they do not support 'response_format', but always return a base64 encoded image.
+         * Any custom options are added to the parameters as well.
+         * This allows developers to pass other options that may be more niche or not yet supported by the SDK.
          */
-        if (isset($params['model']) && is_string($params['model']) && str_starts_with($params['model'], 'gpt-image-')) {
-            unset($params['response_format']);
-        } else {
-            unset($params['output_format']);
+        $customOptions = $config->getCustomOptions();
+        foreach ($customOptions as $key => $value) {
+            if (isset($params[$key])) {
+                throw new InvalidArgumentException(
+                    sprintf(
+                        'The custom option "%s" conflicts with an existing parameter.',
+                        $key
+                    )
+                );
+            }
+            $params[$key] = $value;
         }
 
         return $params;
     }
+
+    /**
+     * Gets the host model to use for image generation requests.
+     *
+     * The image_generation tool runs within a host model's context. For dedicated
+     * image generation models like gpt-image-1, we use a capable host model.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $modelId The requested model ID.
+     * @return string The host model ID to use for the request.
+     */
+    protected function getHostModelForImageGeneration(string $modelId): string
+    {
+        // If this is a gpt-image-* model, we need a host model to run the tool.
+        // Otherwise, the model itself can host the image_generation tool.
+        if (str_starts_with($modelId, 'gpt-image-')) {
+            return 'gpt-4o';
+        }
```

Comment:
Related to the above, this is not great. Since it's a tool and not multimodal output, I think we should expose it as a tool.

Or, at the very least, we should continue to offer an actual image generation model to prompt directly, via the existing `images/generations` endpoint from `AbstractOpenAiCompatibleImageGenerationModel`, without going through the Responses API - which for the scenario of purely image generation seems like an unnecessary proxy in between.